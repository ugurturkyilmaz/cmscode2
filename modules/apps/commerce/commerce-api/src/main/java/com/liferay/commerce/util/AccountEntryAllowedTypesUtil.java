/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.util;

import com.liferay.account.constants.AccountConstants;
import com.liferay.commerce.account.constants.CommerceAccountConstants;

/**
 * @author Luca Pellizzon
 */
public class AccountEntryAllowedTypesUtil {

	public static String[] getAllowedTypes(int commerceSiteType) {
		if (commerceSiteType == CommerceAccountConstants.SITE_TYPE_B2B) {
			return new String[] {AccountConstants.ACCOUNT_ENTRY_TYPE_BUSINESS};
		}

		if (commerceSiteType == CommerceAccountConstants.SITE_TYPE_B2C) {
			return new String[] {AccountConstants.ACCOUNT_ENTRY_TYPE_PERSON};
		}

		if (commerceSiteType == CommerceAccountConstants.SITE_TYPE_B2X) {
			return new String[] {
				AccountConstants.ACCOUNT_ENTRY_TYPE_BUSINESS,
				AccountConstants.ACCOUNT_ENTRY_TYPE_PERSON
			};
		}

		return AccountConstants.ACCOUNT_ENTRY_TYPES_DEFAULT_ALLOWED_TYPES;
	}

}