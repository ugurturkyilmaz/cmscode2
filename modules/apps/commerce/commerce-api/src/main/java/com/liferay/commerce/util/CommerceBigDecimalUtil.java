/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.util;

import java.math.BigDecimal;

/**
 * @author Igor Beslic
 */
public class CommerceBigDecimalUtil {

	public static boolean eq(BigDecimal value1, BigDecimal value2) {
		if (value1.compareTo(value2) == 0) {
			return true;
		}

		return false;
	}

	public static boolean gt(BigDecimal value1, BigDecimal value2) {
		if (value1 == null) {
			return false;
		}

		if (value1.compareTo(value2) > 0) {
			return true;
		}

		return false;
	}

	public static boolean gte(BigDecimal value1, BigDecimal value2) {
		if (value1 == null) {
			return false;
		}

		if (value1.compareTo(value2) >= 0) {
			return true;
		}

		return false;
	}

	public static boolean isZero(BigDecimal value) {
		if ((value == null) || (value.compareTo(BigDecimal.ZERO) == 0)) {
			return true;
		}

		return false;
	}

	public static boolean lt(BigDecimal value1, BigDecimal value2) {
		if (value1.compareTo(value2) < 0) {
			return true;
		}

		return false;
	}

	public static boolean lte(BigDecimal value1, BigDecimal value2) {
		if (value1.compareTo(value2) <= 0) {
			return true;
		}

		return false;
	}

}