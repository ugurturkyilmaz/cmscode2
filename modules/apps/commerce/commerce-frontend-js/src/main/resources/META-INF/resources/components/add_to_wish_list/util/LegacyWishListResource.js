/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {fetch} from 'frontend-js-web';

const RESOURCE_ENDPOINT = '/o/commerce-ui/wish-list-item';

const LegacyWishListResource = {

	/**
	 * @param accountId (required)
	 * @param cpDefinitionId (required)
	 * @param skuId (optional)
	 * @returns {Promise<any | {success: boolean}>}
	 */
	toggleInWishList({accountId = '0', cpDefinitionId, skuId = '0'}) {
		if (!cpDefinitionId) {
			return Promise.resolve({success: false});
		}

		const formData = new FormData();

		formData.append('commerceAccountId', accountId);
		formData.append('groupId', Liferay.ThemeDisplay.getScopeGroupId());
		formData.append('productId', cpDefinitionId);
		formData.append('skuId', skuId);
		formData.append('options', '[]');

		return fetch(RESOURCE_ENDPOINT, {
			body: formData,
			credentials: 'include',
			headers: new Headers({'x-csrf-token': window.Liferay.authToken}),
			method: 'POST',
		}).then((response) => response.json());
	},
};

export default LegacyWishListResource;
