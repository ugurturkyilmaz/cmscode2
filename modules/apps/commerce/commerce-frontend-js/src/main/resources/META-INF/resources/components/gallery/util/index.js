/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {fetch} from 'frontend-js-web';

export function fetchImage(url) {
	return new Promise((resolve) => {
		const image = window.document.createElement('img');

		image.onload = () => {
			resolve(url);
		};

		image.src = url;
	});
}

export function updateGallery(formFields, namespace, viewCPAttachmentURL) {
	const ddmFormValues = JSON.stringify(formFields);
	const formData = new FormData();

	formData.append(`${namespace}ddmFormValues`, ddmFormValues);
	formData.append('groupId', themeDisplay.getScopeGroupId());

	return fetch(viewCPAttachmentURL, {
		body: formData,
		headers: new Headers({'x-csrf-token': Liferay.authToken}),
		method: 'post',
	}).then((response) => response.json());
}
