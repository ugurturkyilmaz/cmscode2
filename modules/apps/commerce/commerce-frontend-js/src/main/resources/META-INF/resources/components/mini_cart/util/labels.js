/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {
	ADD_PRODUCT,
	ORDER_IS_EMPTY,
	REMOVE_ALL_ITEMS,
	REVIEW_ORDER,
	SUBMIT_ORDER,
	VIEW_DETAILS,
	YOUR_ORDER,
} from './constants';

export const DEFAULT_LABELS = {
	[ADD_PRODUCT]: Liferay.Language.get('add-a-product-to-the-cart'),
	[ORDER_IS_EMPTY]: Liferay.Language.get('your-order-is-empty'),
	[REMOVE_ALL_ITEMS]: Liferay.Language.get('remove-all-items'),
	[REVIEW_ORDER]: Liferay.Language.get('review-order'),
	[SUBMIT_ORDER]: Liferay.Language.get('submit'),
	[VIEW_DETAILS]: Liferay.Language.get('view-details'),
	[YOUR_ORDER]: Liferay.Language.get('your-order'),
};
