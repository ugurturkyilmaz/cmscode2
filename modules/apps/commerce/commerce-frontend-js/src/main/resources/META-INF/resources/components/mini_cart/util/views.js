/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {getJsModule} from '../../../utilities/modules';
import Summary from '../../summary/Summary';
import CartItem from '../CartItem';
import CartItemsList from '../CartItemsList';
import CartItemsListActions from '../CartItemsListActions';
import Header from '../Header';
import Opener from '../Opener';
import OrderButton from '../OrderButton';
import Wrapper from '../Wrapper';
import {
	CART,
	HEADER,
	ITEM,
	ITEMS_LIST,
	ITEMS_LIST_ACTIONS,
	OPENER,
	ORDER_BUTTON,
	SUMMARY,
} from './constants';

export const DEFAULT_VIEWS = {
	[CART]: {component: Wrapper},
	[HEADER]: {component: Header},
	[ITEM]: {component: CartItem},
	[ITEMS_LIST]: {component: CartItemsList},
	[ITEMS_LIST_ACTIONS]: {component: CartItemsListActions},
	[OPENER]: {component: Opener},
	[ORDER_BUTTON]: {component: OrderButton},
	[SUMMARY]: {component: Summary},
};

/**
 * decorateWithName - for test purposes only
 * @param componentFn: React [Function] component
 * @param keyValuePairs: object
 */
function decorateWith(componentFn, keyValuePairs) {
	const component = componentFn;

	component.component = {...keyValuePairs};

	return component;
}

function resolveView({component, contentRendererModuleUrl}) {
	if (component) {
		return Promise.resolve(
			decorateWith((props) => component(props), {name: component.name})
		);
	}

	return getJsModule(contentRendererModuleUrl).then((module) =>
		Promise.resolve(
			decorateWith(module, {
				moduleURL: contentRendererModuleUrl,
				name: module.name,
			})
		)
	);
}

export function resolveCartViews(customViews = {}) {
	const views = {...DEFAULT_VIEWS, ...customViews};
	const [...viewTypes] = Object.keys(DEFAULT_VIEWS).sort();

	return Promise.all(
		viewTypes.map((viewType) =>
			resolveView(views[viewType]).catch(() =>
				resolveView(DEFAULT_VIEWS[viewType])
			)
		)
	).then((resolvedViews) =>
		Promise.resolve(
			viewTypes.reduce(
				(views, type) => ({
					...views,
					[type]: resolvedViews.shift(),
				}),
				{}
			)
		)
	);
}
