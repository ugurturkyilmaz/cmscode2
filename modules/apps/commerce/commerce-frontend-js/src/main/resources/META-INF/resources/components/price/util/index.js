/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

const DISCOUNT_LEVEL_PREFIX = 'discountPercentageLevel';

export function adaptLegacyPriceModel(priceModel) {
	if (`${DISCOUNT_LEVEL_PREFIX}1` in priceModel) {
		return priceModel;
	}

	const {
		discountPercentage,
		discountPercentages,
		finalPrice,
		price,
		promoPrice,
	} = priceModel;

	return {
		discountPercentage: parseFloat(discountPercentage || 0),
		finalPriceFormatted: finalPrice || price,
		priceFormatted: price,

		/**
		 * The following matches numbers in the
		 * string value (e.g. "% 90.00" => "9")
		 * and is solely needed to check
		 * that the promoPrice value is non-null.
		 *
		 * Then the promoPriceFormatted must be used.
		 */
		promoPrice: promoPrice ? promoPrice.match(/\d/gi)[0] : '0',
		promoPriceFormatted: promoPrice,
		...(discountPercentages || ['0', '0', '0', '0']).reduce(
			(discountLevels, percentage, i) => ({
				...discountLevels,
				[`${DISCOUNT_LEVEL_PREFIX}${i + 1}`]: parseFloat(percentage),
			}),
			{}
		),
	};
}

export function collectDiscountLevels(price) {
	return Object.keys(price).reduce((levels, key) => {
		if (key.startsWith(DISCOUNT_LEVEL_PREFIX)) {
			levels.push(price[key].toFixed(2));
		}

		return levels;
	}, []);
}

export function isNonnull(...values) {
	return !!values.find((value) => parseFloat(value) > 0);
}
