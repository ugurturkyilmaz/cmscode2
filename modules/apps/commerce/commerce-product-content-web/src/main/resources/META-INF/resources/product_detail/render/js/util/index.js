/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

function fillField(fieldElement) {
	if (fieldElement.jsonData && fieldElement instanceof HTMLElement) {
		const textElement = fieldElement.querySelector('span:nth-child(2)');

		if (textElement) {
			textElement.innerText = Liferay.Util.escape(fieldElement.jsonData);

			fieldElement.classList.remove('hide');
		}
	}
}

export function updateProductFields(data) {
	const cpInstance = data.cpInstance;

	const availabilityEstimateContainer =
		document.querySelector(
			'[data-text-cp-instance-availability-estimate]'
		) || {};
	const gtinContainer =
		document.querySelector('[data-text-cp-instance-gtin]') || {};
	const mpnContainer =
		document.querySelector(
			'[data-text-cp-instance-manufacturer-part-number]'
		) || {};
	const skuContainer =
		document.querySelector('[data-text-cp-instance-sku]') || {};
	const stockQuantityContainer =
		document.querySelector('[data-text-cp-instance-stock-quantity]') || {};

	availabilityEstimateContainer.jsonData = cpInstance.availabilityEstimate;
	gtinContainer.jsonData = cpInstance.gtin;
	mpnContainer.jsonData = cpInstance.manufacturerPartNumber;
	skuContainer.jsonData = cpInstance.sku;

	[
		availabilityEstimateContainer,
		gtinContainer,
		mpnContainer,
		skuContainer,
	].forEach(fillField);

	stockQuantityContainer.innerHTML = cpInstance.stockQuantity
		? cpInstance.stockQuantity + Liferay.Language.get('in-stock')
		: '';

	document.querySelector(
		'[data-text-cp-instance-subscription-info]'
	).innerHTML = cpInstance.subscriptionInfo || '';
	document.querySelector(
		'[data-text-cp-instance-delivery-subscription-info]'
	).innerHTML = cpInstance.deliverySubscriptionInfo || '';
}
