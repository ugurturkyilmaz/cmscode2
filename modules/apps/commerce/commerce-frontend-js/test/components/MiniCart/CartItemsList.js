/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import '@testing-library/jest-dom/extend-expect';
import {cleanup, render} from '@testing-library/react';
import React from 'react';

import CartItemsList from '../../../src/main/resources/META-INF/resources/components/mini_cart/CartItemsList';
import MiniCartContext from '../../../src/main/resources/META-INF/resources/components/mini_cart/MiniCartContext';
import {
	ADD_PRODUCT,
	ITEM,
	ITEMS_LIST,
	ITEMS_LIST_ACTIONS,
	SUMMARY,
} from '../../../src/main/resources/META-INF/resources/components/mini_cart/util/constants';
import {DEFAULT_LABELS} from '../../../src/main/resources/META-INF/resources/components/mini_cart/util/labels';

describe('MiniCart Items List', () => {
	const BASE_CONTEXT_MOCK = {
		CartViews: {
			[ITEMS_LIST_ACTIONS]: () => <div>{ITEMS_LIST_ACTIONS}</div>,
		},
		cartState: {},
		labels: DEFAULT_LABELS,
	};

	const COMPONENT_SELECTOR = '.mini-cart-items-list';

	afterEach(() => {
		jest.resetAllMocks();

		cleanup();
	});

	describe('by default', () => {
		it(
			`renders the ${ITEMS_LIST} component w/ ` +
				`the ${ITEMS_LIST_ACTIONS} and a div including ` +
				`a shopping cart icon and the label "${ADD_PRODUCT}"`,
			() => {
				const EMPTY_CART_SELECTOR = '.empty-cart';

				const {container, getByText} = render(
					<MiniCartContext.Provider value={BASE_CONTEXT_MOCK}>
						<CartItemsList />
					</MiniCartContext.Provider>
				);

				expect(getByText(ITEMS_LIST_ACTIONS)).toBeInTheDocument();

				const CartItemsListElement = container.querySelector(
					COMPONENT_SELECTOR
				);

				const EmptyCartElement = CartItemsListElement.querySelector(
					EMPTY_CART_SELECTOR
				);

				expect(CartItemsListElement).toBeInTheDocument();
				expect(EmptyCartElement).toBeInTheDocument();

				expect(
					getByText(BASE_CONTEXT_MOCK.labels[ADD_PRODUCT])
				).toBeInTheDocument();

				expect(CartItemsListElement.innerHTML).toMatchSnapshot();
			}
		);
	});

	describe('by data flow', () => {
		describe('if there are cart items', () => {
			const WITH_ITEMS_CONTEXT_MOCK = {
				...BASE_CONTEXT_MOCK,
				CartViews: {
					...BASE_CONTEXT_MOCK.CartViews,
					[ITEM]: (props) => <div>{props.item.name}</div>,
					[SUMMARY]: ({dataMapper, isLoading, summaryData}) => {
						dataMapper();

						return (
							<div>
								{isLoading
									? `${SUMMARY} is loading`
									: `${SUMMARY} ${summaryData.itemsQuantity}`}
							</div>
						);
					},
				},
				cartState: {
					cartItems: [
						{
							id: 1,
							name: 'An Item',
						},
					],
					summary: {
						itemsQuantity: 1,
					},
				},
				isUpdating: false,
				labels: DEFAULT_LABELS,
				summaryDataMapper: jest.fn(),
			};

			it(
				`renders the ${ITEMS_LIST} component displaying ` +
					`the ${ITEMS_LIST_ACTIONS}, a div including ` +
					`the list of cart items and the order ${SUMMARY}`,
				() => {
					const {container, getByText} = render(
						<MiniCartContext.Provider
							value={WITH_ITEMS_CONTEXT_MOCK}
						>
							<CartItemsList />
						</MiniCartContext.Provider>
					);

					expect(getByText(ITEMS_LIST_ACTIONS)).toBeInTheDocument();

					const CartItemsListElement = container.querySelector(
						COMPONENT_SELECTOR
					);
					const CartItemElements = CartItemsListElement.querySelectorAll(
						'.mini-cart-cart-items div'
					);

					expect(CartItemElements.length).toEqual(1);
					expect(CartItemElements[0].innerHTML).toEqual(
						WITH_ITEMS_CONTEXT_MOCK.cartState.cartItems[0].name
					);

					expect(
						getByText(
							`${SUMMARY} ${WITH_ITEMS_CONTEXT_MOCK.cartState.summary.itemsQuantity}`
						)
					).toBeInTheDocument();
				}
			);

			it(`if the cart is updating, the ${SUMMARY} will be informed`, () => {
				const {getByText} = render(
					<MiniCartContext.Provider
						value={{
							...WITH_ITEMS_CONTEXT_MOCK,
							isUpdating: true,
						}}
					>
						<CartItemsList />
					</MiniCartContext.Provider>
				);

				expect(getByText(`${SUMMARY} is loading`)).toBeInTheDocument();
			});
		});
	});
});
