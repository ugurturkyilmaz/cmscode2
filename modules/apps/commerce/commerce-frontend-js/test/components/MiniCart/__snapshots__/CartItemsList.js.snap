// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniCart Items List by default renders the ItemsList component w/ the ItemsListActions and a div including a shopping cart icon and the label "Add a product to the cart" 1`] = `"<div>ItemsListActions</div><div class=\\"empty-cart\\"><div class=\\"empty-cart-icon mb-3\\"><svg class=\\"lexicon-icon lexicon-icon-shopping-cart\\" role=\\"presentation\\"><use xlink:href=\\"#shopping-cart\\"></use></svg></div><p class=\\"empty-cart-label\\">add-a-product-to-the-cart</p></div>"`;
